### ISSUE TITLE
PG::UndefinedColumn: ERROR: column host_classes.host_puppet_facet_id does not exist

### ISSUE THREAD

**Issue by jhd85** [theforeman/foreman_puppet#181]
Not sure if this has to do with the plugin of foreman itself. But I did an update to 2.5 and as noted I tried installing the foreman_puppet as it will be removed from core in the future.

Since I did not see the plugin listed with 'foreman-installer' I used the manual install with `yum install tfm-rubygem-foreman_puppet`. As soon as I installed the plugin I get:

```
Oops, we're sorry but something went wrong PG::UndefinedColumn: ERROR: column host_classes.host_puppet_facet_id does not exist LINE 1: SELECT "host_classes".* FROM "host_classes" WHERE "host_clas... ^ 
```

When:
* I try to edit a host
* I go to config->classes
* Probably more

A migrate gives me error:
```
== 20201125113903 MigratePuppetclassesToFacets: migrating =====================
-- add_reference(:host_classes, :host_puppet_facet, {:foreign_key=>true, :index=>true})
   -> 0.0191s
-- add_reference(:hostgroup_classes, :hostgroup_puppet_facet, {:foreign_key=>true, :index=>true})
   -> 0.0125s
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:
<snip>
Caused by:
NoMethodError: undefined method `id' for nil:NilClass
<snip>
```

Any idea what could be the issue? If you need any other info, let me know.

**Comment by ezr-ondrej** [comment]
I think this is being fixed in #175, you've got orpaned classes (classes assigned to host without Env).
I think it is this line: https://github.com/theforeman/foreman_puppet/pull/175/files#diff-6d6617b1ab5115e68a304536691c45bc69c2c3a5a8570e65898d520c93939561L17

Could you apply #175 on your instance (on the plugins - `foreman_puppet` gem - migration) and test that out?

**Comment by jhd85** [comment]
Yes, it seems to work now. Thanks. But I noticed I no longer get puppet reports in foreman now. No idea if it is related. Will look into it later.
